crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals_is.md

```diff
@@ -25,3 +25,17 @@ if y is x:
 
 reveal_type(y)  # revealed: A | None
 ```
+
+## `is` in chained comparisons
+
+```py
+x = True if x_flag else False
+y = True if y_flag else False
+
+reveal_type(x)  # revealed: bool
+reveal_type(y)  # revealed: bool
+
+if y is x is False:  # Interpreted as `(y is x) and (x is False)`
+    reveal_type(x)  # revealed: Literal[False]
+    reveal_type(y)  # revealed: bool
```

Comment:
Technically, given any single transitive relation (and `is` is transitive; `is not`, of course, is not 😆 ) used across all comparisons, any narrowing that applies in one comparison can be applied across all of them; it would be safe to infer `Literal[False]` for `y` here also.

A more general framing would be to say that it's valid to use the type of a name as narrowed in one chained comparison when evaluating another chained comparison involving the same name, so in this case it would be valid to use the type `Literal[False]` for `x` when evaluating narrowing for `y is x`, giving us `Literal[False]` for `y` also. The tricky thing about actually implementing this would be deciding in which order to evaluate the narrowings.

All that said: I don't think it's important that we _do_ this. Perhaps not ever, and certainly not in this PR.

I do think it's important that you also added the "is not" test below, since that one was more clearly wrong with the prior behavior.